src/TypeShape.SourceGenerator/Helpers/RoslynHelpers.AdditionalMarkerAttribute.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace TypeShape.SourceGenerator.Helpers;
+
+internal static partial class RoslynHelpers
+{
+    public static void RegisterAdditionalMarkerAttributeName(this IncrementalGeneratorInitializationContext context,
+        MarkerAttributeHolder holder)
+    {
+        var valueProvider = context.AnalyzerConfigOptionsProvider
+            .Select((options, _) =>
+            {
+                if (options.GlobalOptions.TryGetValue(
```

Comment:
Assuming this gains traction, it might be necessary to support multiple attributes. Would that be possible, e.g. using multiple entries or comma-separated values?